File: python/day1.py
import functools

with open("../_input/day1.txt", "r") as f:
    list1, list2 = zip(*[line.strip().split("   ") for line in f.readlines()])

list1 = sorted(map(int, list1))
list2 = sorted(map(int, list2))

result1 = sum(abs(x1 - x2) for x1, x2 in zip(list1, list2))
result2 = functools.reduce(lambda s, i: s + i * list2.count(i), list1, 0)
# result2 = sum(i * list2.count(i) for i in list1)

print(result1)
print(result2)

File: ruby/day1.rb
list1, list2 = File
  .readlines("../_input/day1.txt")
  .map { |l| l.split("   ") }
  .transpose

list1.map!(&:to_i)
list2.map!(&:to_i)

result1 = list1.sort.zip(list2.sort).map { |x1, x2| (x1.to_i - x2.to_i).abs }.sum
result2 = list1.inject(0) { |s, i| s + i * list2.count(i) }
# result2 = list1.map { |i| i * list2.count(i) }.sum

puts result1
puts result2
